File: DevExpress VCL/ExpressQuantumTreeList 4/Demos/CBuilder/BandedFixedDemo/BandedFixedDemoData.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "BandedFixedDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxStyles"
#pragma link "cxTL"
#pragma resource "*.dfm"
TBandedFixedDemoDataDM *BandedFixedDemoDataDM;
//---------------------------------------------------------------------------
__fastcall TBandedFixedDemoDataDM::TBandedFixedDemoDataDM(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TBandedFixedDemoDataDM::tblShedulerCalcFields(TDataSet *DataSet)
{
  tblShedulerWeekSum->Value = tblShedulerSUNDAY->Value + tblShedulerMONDAY->Value +
    tblShedulerTUESDAY->Value + tblShedulerWEDNESDAY->Value + tblShedulerTHURSDAY->Value +
    tblShedulerFRIDAY->Value + tblShedulerSATURDAY->Value;
  tblShedulerWeekAVG->Value = tblShedulerWeekSum->Value/7;
}
//---------------------------------------------------------------------------

String TBandedFixedDemoDataDM::GetProjectNameByID(int AProjectID)
{
  return tblProjects->Lookup("ID", AProjectID, "Name");
}
//---------------------------------------------------------------------------

String TBandedFixedDemoDataDM::GetPersonNameByID(int APersonID)
{
  return tblPersons->Lookup("ID", APersonID, "Name");
}
//---------------------------------------------------------------------------

File: DevExpress VCL/ExpressQuantumTreeList 4/Demos/Delphi/BandedFixedDemo/BandedFixedDemoData.pas
unit BandedFixedDemoData;

{$I ..\cxVer.inc}

interface

uses
  SysUtils, cxClasses, cxStyles, cxTL, DB, Classes
  {$IFDEF DELPHI6}, Variants{$ENDIF}, DBTables
  {$IFNDEF DELPHI6}, Forms{$ENDIF};

type
  TBandedFixedDemoDataDM = class(TDataModule)
    DataBase: TDataBase;
    tblSheduler: TTable;
    dsSheduler: TDataSource;
    dsPersons: TDataSource;
    tblPersons: TTable;
    tblPersonsID: TIntegerField;
    tblPersonsName: TStringField;
    tblPersonsAddress: TStringField;
    tblPersonsPhone: TStringField;
    tblPersonsFax: TStringField;
    tblPersonsEMAIL: TStringField;
    StyleRepository: TcxStyleRepository;
    cxStyle1: TcxStyle;
    cxStyle2: TcxStyle;
    cxStyle3: TcxStyle;
    cxStyle4: TcxStyle;
    cxStyle5: TcxStyle;
    cxStyle6: TcxStyle;
    cxStyle7: TcxStyle;
    cxStyle8: TcxStyle;
    cxStyle9: TcxStyle;
    cxStyle10: TcxStyle;
    cxStyle11: TcxStyle;
    cxStyle12: TcxStyle;
    cxStyle13: TcxStyle;
    TreeListStyleSheetDevExpress: TcxTreeListStyleSheet;
    tblShedulerID: TAutoIncField;
    tblShedulerPROJECTID: TIntegerField;
    tblShedulerUSERID: TIntegerField;
    tblShedulerSUNDAY: TSmallintField;
    tblShedulerMONDAY: TSmallintField;
    tblShedulerTUESDAY: TSmallintField;
    tblShedulerWEDNESDAY: TSmallintField;
    tblShedulerTHURSDAY: TSmallintField;
    tblShedulerFRIDAY: TSmallintField;
    tblShedulerSATURDAY: TSmallintField;
    dsProjects: TDataSource;
    tblProjects: TTable;
    tblProjectsID: TAutoIncField;
    tblProjectsNAME: TStringField;
    tblProjectsMANAGERID: TIntegerField;
    tblShedulerProjectManagerID: TIntegerField;
    stlGroupNode: TcxStyle;
    tblShedulerWeekSum: TIntegerField;
    tblShedulerWeekAVG: TFloatField;
    stlFixedBand: TcxStyle;
    procedure tblShedulerCalcFields(DataSet: TDataSet);
  public
    { Public declarations }
    function GetProjectNameByID(AProjectID: Integer): string;
    function GetPersonNameByID(APersonID: Integer): string;
  end;

var
  BandedFixedDemoDataDM: TBandedFixedDemoDataDM;

implementation

{$R *.dfm}

{ TBandedFixedDemoDataDM }

function TBandedFixedDemoDataDM.GetPersonNameByID(APersonID: Integer): string;
begin
  Result := tblPersons.Lookup('ID', APersonID, 'Name');
end;

function TBandedFixedDemoDataDM.GetProjectNameByID(AProjectID: Integer): string;
begin
  Result := tblProjects.Lookup('ID', AProjectID, 'Name');
end;

procedure TBandedFixedDemoDataDM.tblShedulerCalcFields(DataSet: TDataSet);
begin
  tblShedulerWeekSum.Value := tblShedulerSUNDAY.Value + tblShedulerMONDAY.Value +
    tblShedulerTUESDAY.Value + tblShedulerWEDNESDAY.Value + tblShedulerTHURSDAY.Value +
    tblShedulerFRIDAY.Value + tblShedulerSATURDAY.Value;
  tblShedulerWeekAVG.Value := tblShedulerWeekSum.Value/7;
end;

end.
